packages/typescript-checker/src/grouping/create-groups.ts

```diff
@@ -39,78 +39,48 @@ import { Node } from './node.js';
  * In this function we create the groups of mutants who can be tested at the same time.
  */
 export function createGroups(mutants: Mutant[], nodes: Map<string, Node>): string[][] {
-  const groups: Mutant[][] = [];
-  let mutant = selectNewMutant(mutants, groups);
-
-  // Loop until all the mutants are in a group
-  while (mutant != null) {
-    // Copy the list of mutants who are not in a group
-    const mutantWhoAreNotInAGroup = [...mutants];
-    // The first mutant is always in the group
-    const group = [mutant];
-    const node = nodes.get(mutant.fileName);
-
-    if (node == null) throw new Error('Node not in graph');
-
-    // Fill the ignoreList
-    let nodesToIgnore = node.getAllParentReferencesIncludingSelf();
-
-    // Loop through the nodes who can possibly go in the group
-    for (const mutantSelected of mutantWhoAreNotInAGroup) {
-      const nodeSelected = nodes.get(mutantSelected.fileName);
-
-      if (nodeSelected == null) throw new Error('Node not in graph');
-
-      // Check if parents of node are not in the group
-      const groupNodes = getNodesFromMutants(group, nodes);
-      if (nodeSelectedHasParentsInCurrentGroup(nodeSelected, groupNodes)) continue;
-
-      // See if the node can be in the group
-      if (nodesToIgnore.has(nodeSelected)) continue;
-
-      // Push the group
-      group.push(mutantSelected);
-      // Mutate the original list and remove the mutant put in the group
-      mutants.splice(mutants.indexOf(mutantSelected), 1);
-      // Add to the ignoreList
-      nodesToIgnore = new Set<Node>([...nodesToIgnore, ...nodeSelected.getAllParentReferencesIncludingSelf()]);
+  const groups: string[][] = [];
```

Comment:
Directly keeping groups as a `string` matrix removes the need to convert it later.